### DISCUSSION TITLE
Set RouteExistingFiles to true

### DISCUSSION THREAD

**Discussion by PonchoPowers** [Dotnet-Boxed/Templates#123] [Q&A]
For discussion:
Set RouteExistingFiles to true to increase performance by removing the need to check the file system to see if an existing path exists, and to allow people to create pages under the path "/services" as most websites which provide a service will use this path.

If you are happy to go ahead with this change I'll create a PR. Thanks.

**Comment by PonchoPowers** [comment]
> Why doesn't ASP.NET MVC grab the request? Two reasons. First, there's an option on RouteCollection called RouteExistingFiles. It's set to false by default which causes ASP.NET MVC to automatically skip routing when a file exists on disk.
> 
> However, in my personal experience, File.Exists is a very expensive operation (very is relative…it's expensive because it hits the disk at all).
> 
> Set routes.RouteExistingFiles = true to turn off the File.Exists check and put ALL the pressure for routing on routes. 

http://www.hanselman.com/blog/PlugInHybridsASPNETWebFormsAndASPMVCAndASPNETDynamicDataSideBySide.aspx

**Comment by RehanSaeed** [comment]
Interesting read @Matthew-Bonner. It also says:

> A best (and example of attention to detail) practice would be to put in an IgnoreRoute call for those pages, directories, and/or HttpHanders.

`IgnoreRoute` does not exist in MVC 6 yet for some reason and I think (not sure) I've configured it correctly for MVC 5 but please go ahead and submit a PR if I'm wrong. Much appreciated!

**Comment by RehanSaeed** [comment]
Feel free to re-open this if you find something.
